app/jobs/recalculate_all_loan_health_job.rb

```diff
@@ -0,0 +1,11 @@
+class RecalculateAllLoanHealthJob < ActiveJob::Base
+  queue_as :default
+
+  def perform
+    # Create a job for each loan recalculation.
+    # The granularity makes it easier to see what loans have problems when they happen.
+    Loan.pluck(:id).each do |loan_id|
```

Comment:
The question is: Should we recalculate all loans or just the loans that have ever had a recalculation done. In theory, the results should be the same. In the other case, we could have orphaned LoanHealthChecks, so I decided with this way.